code/WorkInProgress/DrMelonsStuff.dm

```diff
@@ -114,117 +147,167 @@
 		. += "<br><span class='notice'>[reagents.get_description(user,RC_FULLNESS|RC_VISIBLE|RC_SPECTRO)]</span>"
 		return
 
-	on_reagent_change()
-		..()
-		src.overlays = null
+	// using overlay levels so it looks like you're in the bath
+	// we don't use show_submerged_image since we want the head to poke out
+	// MOB_LAYER(-0) =  all other mobs on top of us
+	// MOB_LAYER-0.1 = Bath edge to trim feet
+	// MOB_LAYER-0.2 = Water overlay
+	// MOB_LAYER-0.3 = Occupant
+	// MOB_LAYER-0.4 = Water underlay
 
+	on_reagent_change()
 		if(reagents.total_volume)
-			ENSURE_IMAGE(src.fluid_icon, src.icon_state, "fluid-bathtub")
+			var/image/new_underlay = src.SafeGetOverlayImage("fluid_underlay", 'icons/obj/stationobjs.dmi', "fluid_bathtub", MOB_LAYER - 0.4)
 			var/datum/color/average = reagents.get_average_color()
-			fluid_icon.color = average.to_rgba()
-			src.overlays += fluid_icon
-
-		if(src.myuser)
-			src.overlays += src.myuser
-			// Don't really see the purpose of this; re-add if you believe otherwise
-			// if(reagents.total_volume)
-				// src.overlays += fluid_icon_feet
-
-		src.overlays += bath_edge
+			var/percent_filled = reagents.total_volume / reagents.maximum_volume
+			if (percent_filled < 0.5)
+				average.a = round(average.a * percent_filled * 2)
+			else
+				average.a = average.a + round((255 - average.a) * (percent_filled - 0.5) * 2)
+			new_underlay.color = average.to_rgba()
+			src.UpdateOverlays(new_underlay, "fluid_underlay")
+
+			if (src.occupant) // only update the overlay if we have an occupant
+				var/image/new_overlay = src.SafeGetOverlayImage("fluid_overlay", 'icons/obj/stationobjs.dmi', "fluid_bathtub", MOB_LAYER - 0.2)
+				new_overlay.color = average.to_rgba()
+				src.UpdateOverlays(new_overlay, "fluid_overlay")
+
+				// I don't want to do this here but no where else feels right
+				// maybe a new chem transfer instead? Could be a new knob to tweak for cyrox too?
+				// #define SUBMERGED 4
+				// it would also make future projects of mine easier, so both this is and i am a hack
+				if(reagents.get_reagent_amount("blood") > 200) // arbitrary
+					// reagents don't track DNA, so it's your own blood vOv
```

Comment:
Blood does! or it should.
It's stored in the vague `data` var- here's a snippet from stable mutagen (no, I don't know why that var is called `cheating`)
```
		var/datum/reagent/blood/cheating = holder.reagent_list["blood"]
		if (cheating && istype(cheating.data, /datum/bioHolder))
			src.data = cheating.data
```
should be able to grab the blood from the tub via the same method as above, get the bioHolder, and then pass the bioHolder's `owner` to `add_blood` to add the right DNA. Wait, why doesn't `add_blood` accept a bioHolder as an argument? why do we have to do this roundabout get-the-owner-to-get-the-bioHolder thing?

```
// WHAT THE ACTUAL FUCK IS THIS SHIT
// WHO THE FUCK WROTE THIS
/atom/proc/add_blood(atom/source, var/amount = 5)
```
ahhh ok then

Reply:
:                        )